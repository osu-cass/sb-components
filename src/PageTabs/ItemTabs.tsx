import * as React from "react";

export type Tabs = "viewer" | "rubric" | "information";

export interface Props {
  changedTab: (tab: Tabs) => void;
  selectedTab: Tabs;
}

export interface State {}

export class ItemTabs extends React.Component<Props, State> {
  private select: Tabs;

  constructor(props: Props) {
    super(props);

    this.state = {
      selectedTab: this.props.selectedTab
    };
  }

  onClick = (selectedVal: Tabs) => {
    this.props.changedTab(selectedVal);
  };

  render() {
    return (
      <div className="tabs">
        <div
          className={
            this.props.selectedTab == "viewer"
              ? "selected-tab"
              : "tab-not-selected"
          }
          onClick={() => this.onClick("viewer")}
        >
          Item Viewer
        </div>
        <div
          className={
            this.props.selectedTab == "rubric"
              ? "selected-tab"
              : "tab-not-selected"
          }
          onClick={() => this.onClick("rubric")}
        >
          Rubric and Exemplar
        </div>
        <div
          className={
            this.props.selectedTab == "information"
              ? "selected-tab"
              : "tab-not-selected"
          }
          onClick={() => this.onClick("information")}
        >
          Item Information
        </div>
      </div>
    );
  }
}
