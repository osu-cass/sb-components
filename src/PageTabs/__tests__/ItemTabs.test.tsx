import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemTabs, Tabs } from '../ItemTabs';
import { shallow, mount, render } from 'enzyme';

describe("PageTabs", () => {
    let tabs: Tabs[] = ["viewer", "rubric", "information"];
    const props = {
        changedTab: jest.fn((tab: Tabs) => {return null}),
        selectedTab: tabs[0]
    }

    it("matches snapshot for each tab", () => {
        let wrapper = shallow(<ItemTabs {...props}/>);
        tabs.forEach(tab => {
            wrapper.setProps({selectedTab: tab});
            expect(wrapper).toMatchSnapshot();
        })
    })

    it("changes tabs", () => {
        let wrapper = shallow(<ItemTabs {...props}/>);
        wrapper.find('div.tabs').childAt(0).simulate('click');
        expect(props.changedTab).toHaveBeenCalled();
    })
})