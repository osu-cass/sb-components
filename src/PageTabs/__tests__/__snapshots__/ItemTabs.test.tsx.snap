// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTabs matches snapshot for each tab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTabs
    changedTab={[Function]}
    selectedTab="viewer"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="selected-tab"
          onClick={[Function]}
        >
          Item Viewer
        </div>,
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Rubric and Exemplar
        </div>,
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Item Information
        </div>,
      ],
      "className": "tabs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Viewer",
          "className": "selected-tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Viewer",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rubric and Exemplar",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Rubric and Exemplar",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Information",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Information",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selected-tab"
            onClick={[Function]}
          >
            Item Viewer
          </div>,
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Rubric and Exemplar
          </div>,
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Item Information
          </div>,
        ],
        "className": "tabs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Viewer",
            "className": "selected-tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Viewer",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rubric and Exemplar",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Rubric and Exemplar",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Information",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Information",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PageTabs matches snapshot for each tab 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTabs
    changedTab={[Function]}
    selectedTab="rubric"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Item Viewer
        </div>,
        <div
          className="selected-tab"
          onClick={[Function]}
        >
          Rubric and Exemplar
        </div>,
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Item Information
        </div>,
      ],
      "className": "tabs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Viewer",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Viewer",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rubric and Exemplar",
          "className": "selected-tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Rubric and Exemplar",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Information",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Information",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Item Viewer
          </div>,
          <div
            className="selected-tab"
            onClick={[Function]}
          >
            Rubric and Exemplar
          </div>,
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Item Information
          </div>,
        ],
        "className": "tabs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Viewer",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Viewer",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rubric and Exemplar",
            "className": "selected-tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Rubric and Exemplar",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Information",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Information",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PageTabs matches snapshot for each tab 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTabs
    changedTab={[Function]}
    selectedTab="information"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Item Viewer
        </div>,
        <div
          className="tab-not-selected"
          onClick={[Function]}
        >
          Rubric and Exemplar
        </div>,
        <div
          className="selected-tab"
          onClick={[Function]}
        >
          Item Information
        </div>,
      ],
      "className": "tabs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Viewer",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Viewer",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rubric and Exemplar",
          "className": "tab-not-selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Rubric and Exemplar",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Information",
          "className": "selected-tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Item Information",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Item Viewer
          </div>,
          <div
            className="tab-not-selected"
            onClick={[Function]}
          >
            Rubric and Exemplar
          </div>,
          <div
            className="selected-tab"
            onClick={[Function]}
          >
            Item Information
          </div>,
        ],
        "className": "tabs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Viewer",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Viewer",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rubric and Exemplar",
            "className": "tab-not-selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Rubric and Exemplar",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Information",
            "className": "selected-tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Item Information",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
